### PR TITLE
Update MyVA Pmt-Info E2E-spec

### PR DESCRIPTION
**Pull request by tlei123**
## Description

Update My VA benefit-payments-and-debt.cypress.spec.js:

- Increase defaultCommandTimeout  config to fix flakiness when using cy-tr-reporter/-helper.
- Reorganize contexts to better outline all user-scenarios.
- Add assertions for both shown & hidden elements for all user-scenario.

## Original issue(s)
[n/a] This is VSA-QA continuous-improvement.

## Testing done

- Does NOT impact app functionalities.  Only the spec was changed, and 1 mock data-file added.
- Ran updated spec locally with cy-tr-helper 10 times in a row, and successfully posted results to TestRail.  [See also screenshot below.]

## Acceptance criteria
- [ ] Code-reviewers approved changes.

## Definition of done
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by tlei123** [comment]
@tmitchellgcio , feel free to ping me for a screenshare if you want a detailed explanation of the spec changes.  It's almost a brand-new spec now.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js [hunk 1]
```diff
@@ -83,124 +105,423 @@ describe('The My VA Dashboard - Payments and Debt', () => {
...
-        cy.findByTestId('payment-card').should('exist');
-        cy.findByTestId('deposit-header').should('exist');
-        cy.findByTestId('payment-card-view-history-link').should('exist');
-        cy.findByTestId('manage-direct-deposit-link').should('exist');
 
-        // make the a11y check
-        cy.injectAxeThenAxeCheck();
+      context('and no payments', () => {
+        // No payments received ever:
+        /* eslint-disable va/axe-check-required */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js:185]: This test's ui-state is the same as a previous test which already had an AXE-check.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js [hunk 2]
```diff
@@ -83,124 +105,423 @@ describe('The My VA Dashboard - Payments and Debt', () => {
...
+          cy.findByTestId('learn-va-debt-link').should('not.exist');
+
+          // make the a11y check
+          cy.injectAxeThenAxeCheck();
+        });
+      });
+
+      context('and no payments', () => {
+        // No payments received ever:
+        /* eslint-disable va/axe-check-required */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js:323]: This test's ui-state is the same as a previous test which already had an AXE-check.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js [hunk 3]
```diff
@@ -83,124 +105,423 @@ describe('The My VA Dashboard - Payments and Debt', () => {
...
+          cy.findByTestId('view-payment-history-link').should('not.exist');
+          cy.findByTestId('manage-direct-deposit-link').should('not.exist');
+          cy.findByTestId('learn-va-debt-link').should('not.exist');
+        });
+        /* eslint-enable va/axe-check-required */
+      });
+
+      context('and payments-API-error', () => {
+        // Payments API returns error:
+        /* eslint-disable va/axe-check-required */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js:357]: This test's ui-state is the same as a previous test which already had an AXE-check.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js [hunk 4]
```diff
@@ -83,124 +105,423 @@ describe('The My VA Dashboard - Payments and Debt', () => {
...
+          cy.findByTestId('learn-va-debt-link').should('exist');
+
+          // make the a11y check
+          cy.injectAxeThenAxeCheck();
+        });
+      });
+
+      context('and no payments', () => {
+        // No payments received ever.
+        /* eslint-disable va/axe-check-required */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js:462]: This test's ui-state is the same as a previous test which already had an AXE-check.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js [hunk 5]
```diff
@@ -83,124 +105,423 @@ describe('The My VA Dashboard - Payments and Debt', () => {
...
+          cy.findByTestId('manage-direct-deposit-link').should('not.exist');
+          cy.findByTestId('view-payment-history-link').should('not.exist');
+          cy.findByTestId('learn-va-debt-link').should('not.exist');
+        });
+        /* eslint-enable va/axe-check-required */
+      });
+
+      context('and payments-API-error', () => {
+        // Payments API returns error.
+        /* eslint-disable va/axe-check-required */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-and-debt.cypress.spec.js:496]: This test's ui-state is the same as a previous test which already had an AXE-check.

**Review by it-harrison — APPROVED** [review]

**Review by adamwhitlock1 — APPROVED** [review]

**Pull request merged**